pkg/apis/pipeline/v1alpha1/task_validation.go

```diff
@@ -130,15 +130,15 @@ func (ts *TaskSpec) Validate(ctx context.Context) *apis.FieldError {
 		}
 	}
 
-	if err := v1beta1.ValidateParameterVariables(ts.Steps, ts.Params); err != nil {
+	if err := v1beta1.ValidateParameterVariables(ctx, ts.Steps, ts.Params); err != nil {
 		return err
 	}
 	// Deprecated
 	if err := validateInputParameterVariables(ts.Steps, ts.Inputs, ts.Params); err != nil {
 		return err
 	}
 
-	if err := v1beta1.ValidateResourcesVariables(ts.Steps, ts.Resources); err != nil {
+	if err := v1beta1.ValidateResourcesVariables(ctx, ts.Steps, ts.Resources); err != nil {
```

Comment:
didn't expect changes to v1alpha1, it should be read only

Reply:
The unit tests were failing because they were calling the same function in `v1beta1` that requires `ctx`.

Reply:
aah I see